### PR TITLE
Genomic super signature

### PR DESCRIPTION
**Pull request by mvdbeek**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by mvdbeek** [comment]
Given there are not a lot of other ravModels, I would suggest dropping `genomic_super_signature_downloader.xml` for now.
In parallel we should have a way to say that certain tools or tool input datatypes are only safe if run containerized, since I guess the possible RDS issues are not a concern if run in containers.

File: tools/genomic_super_signature/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+categories:
+- Sequence Analysis
+- RNA
+- Transcriptomics
+description: Interpretation of RNAseq experiments through robust, efficient comparison to public databases
+homepage_url: https://github.com/shbrief/GenomicSuperSignature
+long_description: ''
```
Comment (bgruening) [inline tools/genomic_super_signature/.shed.yml:7]:
```suggestion
long_description: 'Interpretation of RNAseq experiments through robust, efficient comparison to public databases'
```

File: tools/genomic_super_signature/genomic_super_signature.xml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        <requirement type="package" version="1.7.1">r-optparse</requirement>
+        <requirement type="package" version="2.6">r-wordcloud</requirement>
+        <requirement type="package" version="2.22.0">bioconductor-biocstyle</requirement>
+        <requirement type="package" version="2.7.3">r-magick</requirement>
+        <requirement type="package" version="2021d">tzdata</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #set $model = $model.fields.path
+        mkdir out &&
+        TMPDIR="\$PWD" Rscript '$__tool_directory__/gss.R' --input '$input' --model '$model' --method '$method'  --maxFrom '$maxFrom' --level '$level' --scale '$scale' --numOut $numOut --outDir out --toolDir '$__tool_directory__' --validate '$validate' --html '$html'
```
Comment (bgruening) [inline tools/genomic_super_signature/genomic_super_signature.xml:13]: Is TMPDIR really needed here?

File: tools/genomic_super_signature/genomic_super_signature.xml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        </param>
+        <param argument="--level" type="select" label="Output format of validated result" help="max will output the matrix containing only the maximum coefficient. To get the coefficient of all 8 PCs, set this argument to all.">
+            <option value="max">Max</option>
+            <option value="all">All</option>
+        </param>
+        <param argument="--scale" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Normalize rows of datasets?"/>
+        <param argument="--numOut" type="integer" min="1" value="3" label="The number of top validated RAVs to check"/>
+    </inputs>
+    <outputs>
+        <!-- <data name="plots" format="pdf" label="GenomicSuperSignature plots" from_work_dir="gss_outputs.pdf"/> -->
```
Comment (bgruening) [inline tools/genomic_super_signature/genomic_super_signature.xml:40]: remove?

File: data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                <param name="prior_name" type="select" optional="false" label="Published GenomicSuperSignature model">
+                    <option value="C2" selected="True">MSigDB C2 (curated gene sets)</option>
+                    <option value="PLIERpriors">PLIERpriors: bloodCellMarkersIRISDMAP, svmMarkers, and canonicalPathways</option>
+                </param>
+            </when>
+            <when value="url">
+                <param name="user_url" type="text" area="True" value="https://" optional="false" label="URL" />
+            </when>
+            <when value="history">
+                <param name="input_rav" type="data" format="binary" label="RAV Model file" />
```
Comment (bgruening) [inline data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml:68]: binary is odd

File: data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            </when>
+            <when value="server">
+                <param name="server_filename" type="text" value="" optional="false" label="Full path to RAV file on disk" />
```
Comment (bgruening) [inline data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml:71]: File on disk? That should not be included here I assume.

Comment (blankenberg) [reply data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml:71]: This is for the data manager, so the idea here is if the ravmodel already exists within the shared filesystem, then the admin can specify the existing file, with or without symlinking. For data managers, I think this is generally useful, but happy to remove if it is not preferred.

File: data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml (hunk 4 above)
Comment (blankenberg) [reply data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml]: changed to rds, thanks!

File: tools/genomic_super_signature/genomic_super_signature.xml (hunk 2 above)
Comment (blankenberg) [reply tools/genomic_super_signature/genomic_super_signature.xml]: removed, was likely left over from some other changes due to R writing to tool directory (which was readonly on cvmfs)

File: tools/genomic_super_signature/genomic_super_signature.xml (hunk 3 above)
Comment (blankenberg) [reply tools/genomic_super_signature/genomic_super_signature.xml]: removed, thanks!

File: data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml (hunk 5 above)
Comment (blankenberg) [reply data_managers/data_manager_genomic_super_signature_ravmodels/data_manager/data_manager_genomic_super_signature_ravmodels.xml]: In the interest of moving this PR along quickly, the ability to use an existing ravmodel on the server has been removed. We can revisit as necessary.

File: tools/genomic_super_signature/genomic_super_signature.xml [hunk 6]
```diff
@@ -0,0 +1,208 @@
+<tool id="genomic_super_signature" name="GenomicSuperSignature" version="0.1.0" profile="20.01">
```
Comment (bgruening) [inline tools/genomic_super_signature/genomic_super_signature.xml:1]: should the version follow the bioconductor version? Maybe using recommended IUC macros for versions?

**Comment by bgruening** [comment]
Looks good to me, just one small comment.

File: tools/genomic_super_signature/genomic_super_signature.xml (hunk 6 above)
Comment (blankenberg) [reply tools/genomic_super_signature/genomic_super_signature.xml]: Sounds great. Updated tool wrapper to match bioconductor-genomicsupersignature version and use macros.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @mvdbeek and @blankenberg

**Comment by blankenberg** [comment]
Thank you @bgruening !

**Pull request merged**